src/machine/machine_atsamd51.go

```diff
@@ -113,24 +113,21 @@ const (
 // Return the register and mask to enable a given GPIO pin. This can be used to
 // implement bit-banged drivers.
 func (p Pin) PortMaskSet() (*uint32, uint32) {
-	group := uint8(p) >> 5
-	pin_in_group := uint8(p) & 0x1f
+	group, pin_in_group := p.getPinGrouping()
```

Comment:
Why not just implement `PortMaskSet` and `PortMaskClear` and use that below in all pin manipulations?

Reply:
`PortMaskSet` and `PortMaskClear` point to the `OUTSET` and `OUTCLR` registers.  But most of the Pin methods are using other registers, like `IN`, `DIRSET`, `DIRCLR`, `PMUX`, `PINCFG`.

Maybe I'm missing the point.  Could you go into more detail?  Or mock up some code that demonstrates your idea?

Reply:
Hmm, on second thought it'll probably only complicate things. Nevermind.